### PR TITLE
feat: [CDS-80535]: Side nav v2 expand and collapse

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
#### Screenshots

Side nav expand and collapse
https://github.com/harness/harness-core-ui/assets/62028553/7873921d-5128-4a12-8a43-c4102a79baa6

Default side nav collapsed for execution details page

https://github.com/harness/harness-core-ui/assets/62028553/72b94414-01c1-41a1-a597-7013ae6ae39b

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by avinashmadhwani02** [comment]
retrigger titlecheck

File: src/modules/10-common/navigation/SideNavV2/SideNavLink/SideNavLink.module.scss [hunk 1]
```diff
@@ -12,6 +12,24 @@
...
+    }
+  }
+
+  &.collapsed {
+    align-items: center;
+    justify-content: center;
+
+    .dot {
+      height: 4px;
+      width: 4px;
```
Comment (Sagar-Rajput7) [inline src/modules/10-common/navigation/SideNavV2/SideNavLink/SideNavLink.module.scss:29]: NIT: use vars

File: src/modules/10-common/navigation/SideNavV2/SideNavToggleButton/SideNavToggleButton.module.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+.popover {
+  :global {
+    .bp3-transition-container {
+      z-index: 100;
+    }
+    .bp3-popover {
+      z-index: 100;
+    }
+  }
+}
```
Comment (Sagar-Rajput7) [inline src/modules/10-common/navigation/SideNavV2/SideNavToggleButton/SideNavToggleButton.module.scss:53]: NIT: use vars in this file

File: src/modules/45-projects-orgs/components/ScopeSelector/ScopeSelector.tsx [hunk 3]
```diff
@@ -194,7 +201,12 @@ export const ScopeSelector: React.FC<ScopeSelectorProps> = props => {
             }
             onClick={onButtonClick}
           >
-            <Layout.Vertical spacing="xsmall">{renderSelectedScopeButtonContent()}</Layout.Vertical>
+            {isCollapsed ? (
+              <Icon name="nav-project" size={20} />
+            ) : (
+              <Layout.Vertical spacing="xsmall">{renderSelectedScopeButtonContent()}</Layout.Vertical>
+            )}
+            {/* {!isCollapsed && } */}
```
Comment (Sagar-Rajput7) [inline src/modules/45-projects-orgs/components/ScopeSelector/ScopeSelector.tsx:209]: NIT: cleanup

File: src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx [hunk 4]
```diff
@@ -131,6 +135,19 @@ const CDSideNavLinks = (mode: NAV_MODE): React.ReactElement => {
...
       </SideNav.Section>
       <SideNav.SettingsLink mode={mode} module={module} />
+      {sideNavProps?.sideNavState === SIDE_NAV_STATE.EXPANDED && (
+        <Button
+          className={css.launchButton}
+          variation={ButtonVariation.SECONDARY}
+          size={ButtonSize.SMALL}
+          padding={{ left: 'xxsmall', right: 'xxsmall' }}
+          onClick={() => {
+            window.location.href = `/account/${accountId}/dashboard`
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx:145]: do we have any route for this?

**Review by Sagar-Rajput7 — DISMISSED** [review]

File: src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx (hunk 4 above)
Comment (avinashmadhwani02) [reply src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx:145]: No this is CG route.

File: src/modules/10-common/router/RouteWithLayoutV2.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+              if (changeDefaultExperience) {
+                setSideNavCollapsePrefStore(true)
+                setSideNavState(state)
+              } else {
+                setSideNavState(state)
+              }
+            } else {
+              setSideNavState(state || collapseSideNav ? SIDE_NAV_STATE.COLLAPSED : SIDE_NAV_STATE.EXPANDED)
+            }
+          }
```
Comment (rameshx) [inline src/modules/10-common/router/RouteWithLayoutV2.tsx:61]: `if else` statements can be reduced. Something like:

```typescript
if (!disableSideNavCollapse) {
  if (changeDefaultExperience) {
    setSideNavCollapsePrefStore(state === SIDE_NAV_STATE.COLLAPSED)
  }
  setSideNavState(state)
} else {
  setSideNavState(
    state || collapseSideNav
      ? SIDE_NAV_STATE.COLLAPSED
      : SIDE_NAV_STATE.EXPANDED
  )
}

```

File: src/modules/10-common/router/RouteWithLayoutV2.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        },
+        setDisableSideNavCollapse: (disable: boolean) => {
+          setDisableSideNavCollapse(disable)
+        }
```
Comment (rameshx) [inline src/modules/10-common/router/RouteWithLayoutV2.tsx:65]:
```suggestion
        setDisableSideNavCollapse,
```

File: src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx [hunk 7]
```diff
@@ -131,6 +135,19 @@ const CDSideNavLinks = (mode: NAV_MODE): React.ReactElement => {
         </SideNav.Scope>
       </SideNav.Section>
       <SideNav.SettingsLink mode={mode} module={module} />
+      {sideNavProps?.sideNavState === SIDE_NAV_STATE.EXPANDED && (
+        <Button
+          className={css.launchButton}
+          variation={ButtonVariation.SECONDARY}
+          size={ButtonSize.SMALL}
+          padding={{ left: 'xxsmall', right: 'xxsmall' }}
```
Comment (rameshx) [inline src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx:143]: I don't think we have `xxsmall`

Comment (avinashmadhwani02) [reply src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx]: removed padding. it was not required

File: src/modules/10-common/router/RouteWithLayoutV2.tsx (hunk 6 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/router/RouteWithLayoutV2.tsx]: done

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx [hunk 8]
```diff
@@ -172,14 +185,31 @@ const SideNavMainComponent: React.FC<SideNavMainProps> = props => {
...
+        onMouseEnter={e => {
+          if (isCollapsed && e.currentTarget === e.target) {
+            setSideNavState(SIDE_NAV_STATE.EXPANDED)
+          }
+        }}
+      >
         <SideNavHeader />
         {children}
         <SideNavFooter />
+        {/* {!disableSideNavCollapse && <SideNavToggleButton />} */}
```
Comment (mayankVermaHarness) [inline src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx:212]: Nit: remove comment if not needed.

File: src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx [hunk 9]
```diff
@@ -194,7 +224,11 @@ interface SettingsLinkProps {
 const SettingsTitle = (props: { label: keyof StringsMap }): JSX.Element => {
   const { label } = props
   const { getString } = useStrings()
+  const { sideNavState } = useLayoutV2()
 
+  if (sideNavState === SIDE_NAV_STATE.COLLAPSED) {
+    return <></>
```
Comment (mayankVermaHarness) [inline src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx:230]: return null or undefined?

File: src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx [hunk 10]
```diff
@@ -131,6 +135,20 @@ const CDSideNavLinks = (mode: NAV_MODE): React.ReactElement => {
...
       </SideNav.Section>
       <SideNav.SettingsLink mode={mode} module={module} />
+      {sideNavProps?.sideNavState === SIDE_NAV_STATE.EXPANDED && (
+        <Container className={css.flex1}>
+          <Button
+            className={css.launchButton}
+            variation={ButtonVariation.SECONDARY}
+            size={ButtonSize.SMALL}
+            onClick={() => {
+              window.location.href = `/#/account/${accountId}/dashboard`
```
Comment (mayankVermaHarness) [inline src/modules/75-cd/components/CDSideNav/CDSideNavLinks.tsx:145]: Can we put the CG link in a constant?

**Review by mayankVermaHarness — APPROVED** [review]

File: src/modules/10-common/router/RouteWithContext/RouteWithContext.tsx [hunk 11]
```diff
@@ -18,22 +18,27 @@ import type { PAGE_NAME } from '@common/pages/pageContext/PageName'
...
 export interface RouteWithContextProps extends RouterRouteprops {
   licenseRedirectData?: LicenseRedirectProps
   pageName?: PAGE_NAME
+  sideNavState?: SIDE_NAV_STATE
+  disableAuxNav?: boolean
 }
 
 export function RouteWithContext(props: React.PropsWithChildren<RouteWithContextProps>): React.ReactElement {
-  const { children, licenseRedirectData, pageName, ...rest } = props
+  const { children, licenseRedirectData, pageName, sideNavState, disableAuxNav, ...rest } = props
```
Comment (mayankVermaHarness) [inline src/modules/10-common/router/RouteWithContext/RouteWithContext.tsx:31]: `sideNavState` can be given a default value

Comment (avinashmadhwani02) [reply src/modules/10-common/router/RouteWithContext/RouteWithContext.tsx:31]: It's an optional value passed from parent. Default value is already handled in the parent

File: src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx (hunk 9 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx:230]: JSX.Element doesn't support undefined | null. 
I will check this later, may be we should change the return type of method

File: src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx (hunk 8 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/navigation/SideNavV2/SideNavV2.tsx:212]: this is intentional, expand-collapse feature is disabled for now. we will enable it in the future once the requirements are clear.

**Pull request merged**